### ISSUE TITLE
Default folder is still created despite new preferences

### ISSUE THREAD

**Issue by KimHono** [musescore/MuseScore#15567]
**Describe the bug**
I wanted to get rid of the default folder _~/Documents/MuseScore4_ created each time I launch the software.
So in _Edit>Preferences_, I modified the folder path, and moved the default files in the new folder.
But after restarting, the default folder is one more time created in _~/Documents/_ !
Curiously, if I check the preferences, my choices are those awaited.

**To Reproduce**
1. Copy _~/Documents/MuseScore4_ in another place
2. Start MuseScore 4
3. In menu _Edit>Preferences,_ select the new path everywhere;
4. Stop MuseScore 4
5. Delete _~/Documents/MuseScore4_
6. Start MuseScore 4

**Expected behavior**
_~/Documents/_ may never more include the _MuseScore4_ folder

**Platform information**
 - OS: Linux Arch 6.0.12-arch1-1

**Additional context**
- MuseScore 4.0.0

**Comment by aecati** [comment]
This bug has been discussed here, too: [https://musescore.org/en/node/335161](https://musescore.org/en/node/335161)

It is the only reason, I don`t use MuseScore 4, even it has great improvements. This bug destroys my backups.

**Comment by KimHono** [comment]
Thanks @aecati for this link !
This bug is annoying, but on the other hand MuseScore is working well. So I've been using version 4 since it's release.

**Comment by 707ryanmckay707** [comment]
Hi, I'd like to work on this issue, and it looks like the standard practice is to ask someone to be assigned to it. Could I be assigned to this issue? It would be my first one, and I'm excited to give it a shot!

**Comment by 707ryanmckay707** [comment]
Hi, so while working on this issue, I found another issue, #15264, which is a sub problem, for fixing this bug. So I don't know if you wanted to close it or assign it to me etc.

So far I've fixed the Scores, Styles, Templates, Plugins and Soundfont subfolders from auto generating, with them now only generating at startup if the user hasn't set a custom location. Once I've fixed all the sub folders from autogenerating, then the MuseScore4 Documents folder itself will be fixed from autogenerating. The Cloud Scores and VST3 subfolders are the two main ones that I have left to do.

To fix the Cloud Scores subfolder from autogenerating, the user needs to have the option to change that folder location as well, so I'm planning to add that option to Folder Preferences window, as long as that's okay! (This would solve issue #15264)

**Comment by cbjeukendrup** [comment]
Don't worry about the cloud scores folder; I am currently working on a PR where the cloud scores folder will be moved to a more hidden user data location. So for now, you can just disable the automatic creation of this folder, but better not make is customisable.

**Comment by cbjeukendrup** [comment]
Thanks for the heads-up about that other issue, by the way! I've closed it as a duplicate.

**Comment by 707ryanmckay707** [comment]
Cool thanks, and no problem!

**Comment by 707ryanmckay707** [comment]
Hi so I submitted my pull request! Here's just a couple notes:

To fix the Scores folder specifically, I had to do additional changes compared to other 4 fixed folders, so I included extra details on what the changes were to help with the review process. Also, the changes which made fixing the Scores folder possible, fixed the bug where the default Scores path was not displaying in the Folder Preferences window (This didn't come from changing any QML code, but instead from fixing the Settings::Key value it was pulling from for display). 

Also, there was one small additional bug I found, in the same location that I was implementing part of my fix, so I included it in this pull request, just in case you also wanted it as well. I'm happy to remove that extra bug fix and make any other changes as well!
